### ISSUE TITLE
Enhancement: Android:  New colors for scribble mode

### ISSUE THREAD

**Issue by mlerjen** [OpenOrienteering/mapper#1737]
### Steps to reproduce
The realization of issue #1242 is big fun. You can now draw the line features as contours before drawing the area features **in the background**.  Features drawn in background mode will not cover anything else but only filling the voids.

### Actual behaviour
![image](https://user-images.githubusercontent.com/15611653/93875395-307db600-fcd5-11ea-8e04-d673d51c74c2.png)
fig.1. scribble layer, digitized point & line features and base map. 

This leads to field notes as depicted in fig.1. Though, the background & area mode was only applied for green and yellow, while brown and magenta areas were just filled as in the older days.

This is due the actual color palette is not really ready for the progress already made. There are light colors like green and also dark colors like blue and black and inbetweens like red, magenta and brown. This results in many color line&area combinations with to little distinction.

### Expected behaviour
The pen colors must be distinguishable in between each other but all be held a darker tone, while the area colors also must be distinguishable in between each other, but be held in lighter tone all together.
![image](https://user-images.githubusercontent.com/15611653/93876450-28bf1100-fcd7-11ea-937d-5188792d04a0.png)
fig.2. Possible area and line color palette. Just a sketch. Call for people with real graphical talent :-).

Two possible solutions: 
1) Use this combo as the default. The palette changes to the lighter set, the instant you change to the area mode.
2) Pack the combo into a "lighter" -mode in addition to "solid" and "pattern". The palette does not change, but the lighter set will be applied in area mode

**Comment by ollesmaps** [comment]
I am looking forward to try the new scribbling :)
Bigger color choice in Paint-on-template is also mentined in issue #945.

**Comment by lpechacek** [comment]
The code adding the possibility of defining custom colors is sitting in PR#1738. This ticket might be a good place to discuss how we want to change the *defaults*.

**Comment by lpechacek** [comment]
A test build is available from http://liborpechacek.wz.cz/oomapper/OpenOrienteering-Mapper-refs_pull_1738_merge_v20200925.1-Android-armeabi-v7a.apk

**Comment by mlerjen** [comment]
Thanks for the test build. It is great fun to optimize the palette according ones own needs.
![Screenshot_20200925-144313_OpenOrienteering Mapper](https://user-images.githubusercontent.com/15611653/94410986-52b18100-0178-11eb-8ce0-702d59b2e63d.jpg)
fig.1. You get to see the entire dialog if switching to landscape mode before open it.

Two reflections on this. As it is now the palette will only be adapted after a restart of the app.  IMO there should be the possibility to change colors while the app is running. This I see for two cases. 
a) As it is now, I have a palette of 7 colors to compose. If I want to use the full potential of the area features I would have to portion these into 3 resp. 4 line colors and 4 resp. 3 area colors. see fig.2. The non plus ultra though would be to have 7 colors of  each switching from a line to an area color palette or back if changing the pen mode.
![Screenshot_20200926-203454_OpenOrienteering Mapper](https://user-images.githubusercontent.com/15611653/94413364-44b12f80-017b-11eb-8ae8-d4a9754a6b99.jpg) fig.2 Three area colors, four line colors.
b) To allow both a simple and a sophisticated solution at the same time I would propose that we developed  a mature default palette kind like the one proposed above and then allow the user up to two custom color palettes, one could smoothly switch in between.  Though I propose introducing the palette as a container for a set of colors.

**Comment by ollesmaps** [comment]
AMAZING! I can add colours. Pink is there! It is perfect :D
Thank you @lpechacek :1st_place_medal: 

Also the tweak that shows which scribble template you can use is very efficient!
Thank you @dg0yt  :1st_place_medal:

**Comment by lpechacek** [comment]
Now that we have the infrastructure in place, time has come for the actual color definition.

**Comment by mlerjen** [comment]
![image](https://user-images.githubusercontent.com/15611653/106876358-e6bbc800-66d7-11eb-825d-d6fb009267a0.png)

I will test this palette the next days. In this case the palette is intended to be used for hybrid mapping. The colors thus shall also be distinguishable to the IOF symbol colors. Here I work with four line colors (1) and four area colors (2), the neutral background to cover template (3) and the eraser (4).

btw. really cool side effect of the area-eraser.

**Comment by ollesmaps** [comment]
The palette @mlerjen suggests seems pretty good. 
The rule "_colors shall  be distinguishable to the IOF symbol colors_" is a good one.
In that respect I would maybe shift a little the definition of upper blue, green and lower yellow. Blue, green and yellow object drawn in ISOM vectors shall stand out from the scribble layer.

I might be testing later in February, my latest palette here
https://www.facebook.com/ollesmaps/photos/a.164754419675/10160553998984676/

FIY @lpechacek

**Comment by lpechacek** [comment]
Gentlemen, v0.9.5 is close (#1895), and we've got the custom palette functionality in it. But we are missing convenient presets for those uninitiated. I'd be happy to see palettes that you consider useful for the general public, along with the preferred name that would appear in the dialog. I'll transfer your contributions to the code.

FWIW, I'm now using Martin's "000000,FAF607,00A400,A4FF00,0000FE,B890E3,F6260F,FCB9F4,FAEFD2", with the last color changed to brown. I'm using red for vegetation boundaries and brown for ground shapes, and I'm happy with it in the terrain where I work now.
![image](https://user-images.githubusercontent.com/5584406/110448776-cbbaea00-80c1-11eb-97ee-35f2de1dae3e.png)

**Comment by lpechacek** [comment]
> Thus 000000,FAF607,00A400,B4FF00,0000FE,81EEF7,F6260F,FCB9F4,FAEFD2
> That makes me less tired in the field and on desktop just to avoid dealing with abstraction all the time.

Thanks, Martin, for your contribution! I'll push it to Mapper as is. What is the preferred name you would like to see in the configuration dialog, please?

> At the desktop I would check the completeness of the transfer from scribble to map by switching templates off and on ((F10) so I would see missing areas on map blinking.

Yep. I do the same and it works fine. I've started thinking about how to extend CoVe for the vectorization of these areas. Don't hold your breath though.

Just in case someone would like to look into it, the areas to tackle in CoVe are:
1) Vectorize only part of the image. Let the user select an arbitrarily shaped area for vectorization - AKA "todays' work".
2) Give the user possibility to assign color groups to multiple symbols. Now we are vectorizing a group of colors to one symbol.
3) Detect neighboring areas and make sure they share borders. We don't want to see gaps in the thickets.
4) Resolve the effects of antialiasing. Detect the lighter border pixels and blended pixels as part of the corresponding area object.

**Comment by lpechacek** [comment]
> If the community is willing to adopt the term, "Hybrid Mapping Set*" would make sense to me.

Speaking for myself, I like the name. It nicely captures that the palette complements the standard ISOM symbol set and does not serve as a full palette. I'm inclined to call the palette settings "color sets" and for the purpose of the configuration dialog, I propose omitting the noun. Therefore the short name would become "Hybrid Mapping."

> btw. I noticed that as soon I activate one of the two presets, the custom string gets overwritten ...

That is intended. I acknowledge that the internal dialog logic might not be reflected in its layout. Would you mind filing a new ticket for the issue, please?

@ollesmaps
> Would be a Guacamolle an appropriate name?

I'm fine with the name. To better align it with Martin's proposal, and to give the users a hint about what to expect from the color set, I'm for calling it "Guacamole Full" (mind the different spelling). In my understanding, this broad palette gives the possibility to sketch field notes without the need to use vector objects.

To sum it up, there will be:
* Traditional
* Hybrid Mapping
* Guacamole Full

@mlerjen, @ollesmaps, @dg0yt, speak up if you disagree. Tomorrow I'll post the patch. Thanks!

> @lpechacek I hope yours set will be there too!

Not really. I serve as an enabler here, and honestly, I usually adapt to what I have at hand. For example, on my last map, I fluently adopted four different palettes, so I'm not a guy who would try to get kudos in this area. :)

**Comment by dg0yt** [comment]
:rainbow:  :+1:

**Comment by mlerjen** [comment]
Uh. @ollesmaps Guacamole Full builds up a contrast in which "Hybrid Mapping" appears utterly sober and technical. Adapting to his far more playful approach I would end up somewhere like "Confetti Hybrid" or even better if both names could be switched to Full Guacamole & Hybrid Fruit Sallad? 🤣  To playful? - I am fine with whichever...

**Comment by lpechacek** [comment]
I like the direction in which the naming evolves! :) But let's not overdo it. I'll post a patch with:
* Traditional (color set)
* Confetti Hybrid (color set)
* Guacamole Full (color set)

The "color set" is a hint on how to read the names, and it will not appear in the dialog. Many thanks for your contributions!
